lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1753,6 +1857,21 @@ std::unique_ptr<Rule> BuildSystemEngineDelegate::lookupRule(const KeyType& keyDa
       ));
     }
 
+    if (node->isDirectory()) {
+      return std::unique_ptr<Rule>(new BuildSystemRule(
+        keyData,
+        node->getSignature(),
+        /*Action=*/ [node](BuildEngine& engine) -> Task* {
+          return new ProducedDirectoryNodeTask(*node);
+        },
+        /*IsValid=*/ [node](BuildEngine& engine, const Rule& rule,
+                            const ValueType& value) -> bool {
+          return ProducedDirectoryNodeTask::isResultValid(
+              engine, *node, BuildValue::fromData(value));
+        }
+      ));
+    }
+
     // Otherwise, create a task for a produced node.
```

Comment:
Can you move this comment above the additions so we differentiate between produced and non-produced nodes?